charts/redpanda/node_pools.go

```diff
@@ -0,0 +1,748 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package redpanda
+
+import (
+	"fmt"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+
+	"github.com/redpanda-data/redpanda-operator/gotohelm/helmette"
+	redpandav1alpha3 "github.com/redpanda-data/redpanda-operator/operator/api/redpanda/v1alpha3"
+	"github.com/redpanda-data/redpanda-operator/operator/pkg/clusterconfiguration"
+)
+
+func pooledRedpandaConfigMaps(dot *helmette.Dot, pools []*redpandav1alpha3.NodePool) []*corev1.ConfigMap {
+	cms := []*corev1.ConfigMap{}
+	for _, pool := range pools {
+		cms = append(cms, pooledRedpandaConfigMap(dot, pool, pools))
+	}
+	return cms
+}
+
+func pooledRedpandaConfigMap(dot *helmette.Dot, pool *redpandav1alpha3.NodePool, pools []*redpandav1alpha3.NodePool) *corev1.ConfigMap {
+	bootstrap, fixups := BootstrapFile(dot)
+
+	// Per-pool Redpanda configuration - ugly hack requires remarshaling for now
+	redpandaTemplate := RedpandaConfigFile(dot, pools, true /* includeSeedServer */)
+	template := helmette.FromYaml[map[string]any](redpandaTemplate)
+	poolExtra, poolFixes, _ := translateSource(pool.Spec.BrokerTemplate.NodeConfig)
+	template = helmette.Merge(template, poolExtra)
+	redpandaTemplate = helmette.ToYaml(template)
+	poolFixups := helmette.ToYaml(poolFixes)
+	if len(poolFixes) == 0 {
+		poolFixups = `[]`
+	}
+
+	return &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-%s", Fullname(dot), pool.Name),
+			Namespace: dot.Release.Namespace,
+			Labels:    FullLabels(dot),
+		},
+		Data: map[string]string{
+			clusterconfiguration.BootstrapTemplateFile:    bootstrap,
+			clusterconfiguration.BootstrapFixupFile:       fixups,
+			clusterconfiguration.RedpandaYamlTemplateFile: redpandaTemplate,
+			clusterconfiguration.RedpandaYamlFixupFile:    poolFixups,
+		},
+	}
+}
+
+func renderPool(dot *helmette.Dot, pool *redpandav1alpha3.NodePool, pools []*redpandav1alpha3.NodePool) *appsv1.StatefulSet {
```

Comment:
We have to find a way to share this code with the existing chart, this is too much duplication of code I'd rather not even have a single instance of.

We should be able to coerce `Values` into a `NodePool` and then start swapping out instances of dot / values with `NodePool`, no?